s2spy/_plot.py

```diff
@@ -188,8 +188,8 @@ def matplotlib_visualization(
         ax.xaxis.set_major_formatter(formatter)
         ax.set_xlim(
             (
-                calendar.flat.values.min().left - pd.Timedelta(days=14),
-                calendar.flat.values.max().right + pd.Timedelta(days=14),
+                intervals.stack().values.min().left - pd.Timedelta(days=14),
+                intervals.stack().values.max().right + pd.Timedelta(days=14),
```

Comment:
Did this go wrong somehow? Or is this an efficiency improvement (not recalculating the calendar every time)?

Reply:
Yep~ Since `calendar.flat` includes the whole time, when displaying only a few anchor years (e.g. `n_years = 3`) with `relative_dates=False`, the intervals will become super small. So with `intervals.stack()`, the display is adjusted to the number of years to be included, which looks much nicer 😎.